src/main/groovy/org/asciidoctor/gradle/AsciidoctorTask.groovy

```diff
@@ -852,6 +851,40 @@ class AsciidoctorTask extends DefaultTask {
 
     private void deprecated(final String method, final String alternative, final String msg = '') {
         logger.lifecycle "Asciidoctor: ${method} is deprecated and will be removed in a future version. " +
-            "Use ${alternative} instead. ${msg}"
+                "Use ${alternative} instead. ${msg}"
+    }
+
+    @SuppressWarnings('CatchException')
+    private withAsciidoctor(Closure<?> cl) {
+        EXEC_LOCK.lock()
```

Comment:
This is another implicit change.

With this in place, asciidoctor tasks (with the same extension impl and gem classpath), won't execute concurrently. We did it like this in our patch because we'd rather avoid having duplicate JRuby runtimes than have the tasks execute concurrently. In our case, this is faster. This might not be what all users want though.